core/templates/base-components/oppia-footer.component.ts

```diff
@@ -37,43 +37,64 @@ import './oppia-footer.component.css';
   templateUrl: './oppia-footer.component.html',
   styleUrls: ['./oppia-footer.component.css'],
 })
-export class OppiaFooterComponent {
+export class OppiaFooterComponent implements OnDestroy {
   emailAddress: string | null = null;
   name: string | null = null;
   siteFeedbackFormUrl: string = AppConstants.SITE_FEEDBACK_FORM_URL;
   currentYear: number = new Date().getFullYear();
   PAGES_REGISTERED_WITH_FRONTEND = AppConstants.PAGES_REGISTERED_WITH_FRONTEND;
-
   BRANCH_NAME = AppConstants.BRANCH_NAME;
-
   SHORT_COMMIT_HASH = AppConstants.SHORT_COMMIT_HASH;
 
+  private debounceSubject: Subject<void> = new Subject();
+  private unsubscribe$: Subject<void> = new Subject();
   versionInformationIsShown: boolean =
     this.router.url === '/about' && !AppConstants.DEV_MODE;
 
+  isSubmitting: boolean = false;
+  debounceTimeout: number = 1500;
+
   constructor(
     private alertsService: AlertsService,
     private ngbModal: NgbModal,
     private mailingListBackendApiService: MailingListBackendApiService,
-    private platformFeatureService: PlatformFeatureService,
     private router: Router,
     private windowRef: WindowRef,
     private siteAnalyticsService: SiteAnalyticsService
-  ) {}
+  ) {
+    this.debounceSubject
+      .pipe(
+        debounceTime(this.debounceTimeout),
+        switchMap(() => this.subscribeToMailingListInternal()),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe();
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
 
   getOppiaBlogUrl(): string {
     return '/blog';
   }
 
   validateEmailAddress(): boolean {
-    let regex = new RegExp(AppConstants.EMAIL_REGEX);
+    const regex = new RegExp(AppConstants.EMAIL_REGEX);
     return regex.test(String(this.emailAddress));
   }
-
   subscribeToMailingList(): void {
```

Comment:
The naming you are using here is a bit messy. I recommend, for actions taken that correspond to the UI, using something like "on...", e.g. onSubscribeButtonClicked(). Then that can call subscribeToMailingList(), validateEmailAddress(), etc. internally.

Also please give your debounce subject a better name that provides more info around what it's debouncing, otherwise it's a bit like calling a variable "variable" -- it's too vague and doesn't explain the purpose of the item, only what "type" it is.

Reply:
> The naming you are using here is a bit messy. I recommend, for actions taken that correspond to the UI, using something like "on...", e.g. onSubscribeButtonClicked(). Then that can call subscribeToMailingList(), validateEmailAddress(), etc. internally.
> 
> Also please give your debounce subject a better name that provides more info around what it's debouncing, otherwise it's a bit like calling a variable "variable" -- it's too vague and doesn't explain the purpose of the item, only what "type" it is.

@seanlip thanks for the feedback! I completely agree with your suggestions.

I’ll update the naming conventions to something more intuitive and action-based, like onSubscribeButtonClicked(), for clarity on the actions being taken in the UI. That will also help make the code more readable and maintainable.

As for the debounce subject, I see your point. I'll rename it to something more specific, such as emailSubscriptionDebounceSubject, to better reflect its purpose and avoid any vagueness. This should improve the clarity around what exactly is being debounced.

I’ll implement these changes shortly. Thanks again for the guidance!